py/desispec/workflow/exptable.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import os
+import numpy as np
+from astropy.table import Table
+from astropy.io import fits
+## Import some helper functions, you can see their definitions by uncomenting the bash shell command
+from desispec.workflow.os import define_variable_from_environment, pathjoin, give_relevant_details, get_json_dict
+
+
+def get_exposure_table_column_defs(return_default_values=False):
+    ## Define the column names for the exposure table and their respective datatypes, split in two
+    ##     only for readability's sake
+    colnames1 = ['EXPID', 'EXPTIME', 'OBSTYPE', 'SPECTROGRAPHS', 'CAMWORD', 'TILEID']
+    coltypes1 = [int, float, 'S8', 'S10', 'S30', int]
+    coldeflt1 = [-99, 0.0, 'unknown', '0123456789', 'a09123456789', -99]
+
+    colnames2 = ['NIGHT', 'EXPFLAG', 'HEADERERR', 'SURVEY', 'SEQNUM', 'SEQTOT', 'PROGRAM', 'MJD-OBS']
+    coltypes2 = [int, int, np.ndarray, int, int, int, 'S30', float]
+    coldeflt2 = [20000101, 0, np.array([], dtype=str), 0, 1, 1, 'unknown', 50000.0]
+
+    colnames3 = ['REQRA', 'REQDEC', 'TARGTRA', 'TARGTDEC', 'COMMENTS']
+    coltypes3 = [float, float, float, float, np.ndarray]
+    coldeflt3 = [-99.99, -89.99, -99.99, -89.99, np.array([], dtype=str)]
+
+    colnames = colnames1 + colnames2 + colnames3
+    coldtypes = coltypes1 + coltypes2 + coltypes3
+    coldeflts = coldeflt1 + coldeflt2 + coldeflt3
+
+    if return_default_values:
+        return colnames, coldtypes, coldeflts
+    else:
+        return colnames, coldtypes
+
+def default_exptypes_for_exptable():
+    ## Define the science types to be included in the exposure table (case insensitive)
+    return ['arc','flat','twilight','science','sci','dither','dark','bias','zero']
+
+def get_survey_definitions():
```

Comment:
This survey definition and survey number seems a new concept.  What is it used for at the spectro processing level?

Reply:
It is something you and I brainstormed over the summer, but it's still half-formed. Currently it's not used for anything but could be useful in re-processing multiyear data to delineate between e.g. minisv2, SV0, SV, Y1, etc.

Reply:
As mentioned in my last comment, this is still evolving and the definitions are rather arbitrary at this point. The values aren't used for anything at the moment. They could eventually be useful for parsing a large table for data from specific time periods, e.g. commissioning, SV0, SV1, Y1, etc. But I expect this to evolve as we get more data and determine the best use of such a flag.
I added a brief description of the motivation in the exptable.py "datamodel" at the beginning of that file. For now I will let that resolve this issue for the time being and we can revisit it when it becomes more relevant.